src/ray/common/task/task_spec.h

```diff
@@ -17,7 +17,52 @@ extern "C" {
 }
 
 namespace ray {
-typedef ResourceSet SchedulingClassDescriptor;
+
+struct SchedulingClassDescriptor {
+ public:
+  explicit SchedulingClassDescriptor(const ResourceSet resource_set,
+                                     const FunctionDescriptor function_descriptor)
+      : resource_set_(resource_set), function_descriptor_(function_descriptor) {}
+  explicit SchedulingClassDescriptor(const SchedulingClassDescriptor &sched_cls)
+      : resource_set_(sched_cls.resource_set_),
+        function_descriptor_(sched_cls.function_descriptor_) {}
+
+  const std::unordered_map<std::string, double> GetResourceMap() const {
```

Comment:
const ref?

Reply:
Hmm can't seem to do that because that would be returning a reference to a stack object. (the actual resource map is actually being constructed in the function, not just returning an instance variable).